assembl/static2/js/app/components/debate/common/ideasLevel.jsx

```diff
@@ -0,0 +1,408 @@
+
+        setTimeout(goNextStep, 500);
+        return;
+      } else if (step === 1) {
+        /*
+        Animation step 1:
+          - Set ideas positioning to a single row (set row display to inline-flex)
+          - Set necessary values consequently to inline-flex, for animation consistency
+        */
+        this.initializeConstants();
+
+        const row = this.getRow();
+
+        row.style.overflowX = 'hidden';
+        row.style.display = 'inline-flex';
+
+        setTimeout(goNextStep, 50);
+        return;
+      } else if (step === 2) {
+        /*
+        Animation step 2:
+          - Animate row width from full page width to custom value
+          - Animate ideas width from initial width to carousel-adapted width
+        */
+
+        this.setOption('isAnimatingTowardsInline', true);
+        this.updateWidth();
+
+        const row = this.getRow();
+        const thematicWidthPx = `${this.getOption('thematicWidth')}px`;
+        row.childNodes.forEach((el) => {
+          const el2 = el;
+          el2.style.width = thematicWidthPx;
+        });
+
+        this.scrollToIdeaIndexIfNecessary(index);
+
+        const f = () => {
+          this.onSeeSubIdeasClick(ideaId, index, step + 1);
+          this.forceUpdate();
+        };
+        f.bind(this);
+
+        setTimeout(f, 500);
+
+        return;
+      }
+    }
+
+    this.setOption('isAnimatingTowardsInline', true);
+    this.me.classList.add('animating-towards-inline');
+
+    this.setOption('selectedIdea', ideaId);
+
+    this.initializeConstants();
+
+    // make sure that the clicked thematic shows fully when inline (scroll at the correct position so that it is shown)
+    if (!wasInline) {
+      this.scrollToIdeaIndexIfNecessary(index);
+    }
+    this.updateScrollButtonsVisibility(this.state.isInline, true);
+  }
+
+  onScrollRightClick() {
+    this.initializeConstants();
+    this.moveAllChildren(this.getRow(), -1 * this.getOption('animationDistance'));
+  }
+
+  onScrollLeftClick() {
+    this.initializeConstants();
+    this.moveAllChildren(this.getRow(), this.getOption('animationDistance'));
+  }
+
+  getRow() {
+    if (this.me) {
+      return this.me.childNodes[1];
+    }
+    return null;
+  }
+
+  getOption(name) {
+    return getValue(this, ['options', name], null);
+  }
+
+  setOption(name, value) {
+    this.options[name] = value;
+  }
+
+  updateWidth() {
+    if (this.me) {
+      if (this.state.isInline || this.getOption('isAnimatingTowardsInline')) {
+        this.me.style.width = `${this.getOption('carouselWidth')}px`;
+        const row = this.getRow();
+        if (row) {
+          row.style.width = `${this.getOption('carouselWidth')}px`;
+        }
+      }
+    }
+  }
+
+  scrollToIdeaIndexIfNecessary(index) {
+    const thematicWidth = this.getOption('thematicWidth');
+    const carouselWidth = this.getOption('carouselWidth');
+    const ideaInitialX = index * thematicWidth;
+    if (ideaInitialX + thematicWidth > carouselWidth) {
```

Comment:
the else is do nothing (which means don't do any scroll), hence the name of the method scrollToIdeaIndexIfNecessary